server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -898,7 +898,9 @@ public VolumeVO createVolume(CreateVolumeCmd cmd) {
             created = false;
             VolumeInfo vol = volFactory.getVolume(cmd.getEntityId());
             vol.stateTransit(Volume.Event.DestroyRequested);
-            throw new CloudRuntimeException("Failed to create volume: " + volume.getId(), e);
+            String message = String.format("Failed to create volume [%s] due to [%s].", volume.getId(), e.getMessage());
+            s_logger.error(message, e);
+            throw new CloudRuntimeException(message, e);
```

Comment:
why log *and* throw? we can do with either.

Reply:
@DaanHoogland as the real problem was not being logged, I thought it would be better to log it and keep the previous behavior. 

What's your suggestion?

Reply:
A CloudRuntimeException is caught and handled somewhere so I'd just throw. The logging will be hard to find and link to the proper occurrence. If the handling code doesn't log the exception than we should fix that. We potentially get the same stacktrace twice in the log now.
This is no :+1: by any means by the way. I'm just not so keen on overfilling the log with the same info multiple times, It's big enough as is.

Reply:
Seems fair, I will undo the last change.

As the focus of this PR is not the exception handling code, I think we should create an issue to investigate it.